### PR TITLE
chore: add issue analyzer

### PR DESCRIPTION
**Pull request by jrandolf-google**
See internal note.

### CODE REVIEW

File: tools/analyze_issue.mjs [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+  return values
+    .map(value => {
+      return `\`${value}\``;
+    })
+    .join(' ,');
+};
+const formatMessage = value => {
+  return JSON.stringify(value.trim());
+};
+const removeV = value => {
```
Comment (Lightning00Blade) [inline tools/analyze_issue.mjs]: Nit: `removeVPrefix` or `removeVersionPrefix`

File: tools/analyze_issue.mjs [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+'use strict';
+
+import {writeFile, mkdir, copyFile} from 'fs/promises';
+import {dirname, join} from 'path';
+import semver from 'semver';
+import {fileURLToPath} from 'url';
+
+const LAST_SUPPORTED_PUPPETEER_VERSION = '18.2.1';
+const LAST_SUPPORTED_NODE_VERSION = '14.19.0';
```
Comment (Lightning00Blade) [inline tools/analyze_issue.mjs:22]: For the future we should pull this from NPM and Puppeteer repo respectively.

File: tools/analyze_issue.mjs [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+const removeV = value => {
+  return value.startsWith('v') ? value.slice(1) : value;
+};
+
+const ERROR_MESSAGES = {
+  unsupportedOs(value) {
+    return formatMessage(`
+This issue has an unsupported OS: \`${value}\`. Only the following operating systems are supported: ${codifyAndJoinValues(
+      SUPPORTED_OSES
+    )}. Please edit the form with a supported OS.
```
Comment (Lightning00Blade) [inline tools/analyze_issue.mjs]: This sounds like just fix it to work, rather then test it on supported OS. How about:
`Please verify the issue on one the supported OS and update[edit] the issue[form].`

File: tools/analyze_issue.mjs [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+      runsOn = 'windows-latest';
+      break;
+    case 'macos':
+      runsOn = 'macos-latest';
+      break;
+    case 'linux':
+      runsOn = 'ubuntu-latest';
+      break;
+    default:
+      console.log(`errorMessage=${ERROR_MESSAGES.unsupportedOs(os)}`);
```
Comment (Lightning00Blade) [inline tools/analyze_issue.mjs]: Nit: Here and for the other errors as well.
```suggestion
      console.error(`errorMessage=${ERROR_MESSAGES.unsupportedOs(os)}`);
```

File: tools/analyze_issue.mjs [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+      throw new Error(`Unsupported OS: ${os}`);
+  }
+
+  if (!SUPPORTED_PACKAGE_MANAGERS.includes(packageManager)) {
+    console.log(
```
Comment (Lightning00Blade) [inline tools/analyze_issue.mjs]: Looks like there is a lot of duplicate code here. What about a function here (Predicate, ErrorMessage, ThrownError) => void/Throw ?

File: tools/analyze_issue.mjs [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+  const lines = input.split('\n');
+  {
+    /** @type {(value: string) => void} */
+    let set = () => {
+      return void 0;
+    };
+    let j = 1;
+    let i = 1;
+    for (; i < lines.length; ++i) {
+      if (lines[i].startsWith('### Bug behavior')) {
```
Comment (Lightning00Blade) [inline tools/analyze_issue.mjs:111]: I think an Array with {title: "### Bug behavior", execute: (value) => void }. Then looping though it . If it starts we set the `set` and `brake`.

File: tools/analyze_issue.mjs [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+      `errorMessage=${ERROR_MESSAGES.invalidPackageManagerVersion(
+        packageManagerVersion
+      )}`
+    );
+    throw new Error(
+      `Invalid package manager version: ${packageManagerVersion}`
+    );
+  }
+
+  console.log(`errorMessage=""`);
```
Comment (Lightning00Blade) [inline tools/analyze_issue.mjs]: I think we should just add `@actions/code` and use it here rather then using this to output method.

File: tools/analyze_issue.mjs (hunk 4 above)
Comment (jrandolf) [reply tools/analyze_issue.mjs]: We use log here because the message needs to go to `$GITHUB_OUTPUT`

File: tools/analyze_issue.mjs (hunk 7 above)
Comment (jrandolf) [reply tools/analyze_issue.mjs]: What is `@actions/code`?

File: tools/analyze_issue.mjs (hunk 5 above)
Comment (jrandolf) [reply tools/analyze_issue.mjs]: This is more semantic.

File: tools/analyze_issue.mjs (hunk 6 above)
Comment (jrandolf) [reply tools/analyze_issue.mjs:111]: It's the same length of code.

File: tools/analyze_issue.mjs (hunk 7 above)
Comment (Lightning00Blade) [reply tools/analyze_issue.mjs]: For history -> Typo on my part `@actions/core` is package that simplifies working with GitHub Actions

File: tools/assets/verify_issue.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+          process.exit(0);
+        }
+        console.error('Script failed; unknown error found.');
+        process.exit(1);
+      }
+    };
+  }
+
+  for (let i = 0; i < maxRepetitions; ++i) {
+    const result = spawnSync('npm', ['start'], {
```
Comment (OrKoN) [inline tools/assets/verify_issue.ts:51]: do we want a timeout here if the user script never resolves?

Comment (jrandolf) [reply tools/assets/verify_issue.ts:51]: I put it on the step.

**Review by OrKoN — APPROVED** [review]

**Comment by OrKoN** [comment]
Let's wait for @Lightning00Blade's LGTM.

**Review by Lightning00Blade — APPROVED** [review]

**Pull request merged**